git_machete/client.py

```diff
@@ -82,13 +82,13 @@ def __raise_no_branches_error(self) -> None:
             f"machete discover` or `git machete edit`, or edit"
             f" {self._definition_file_path} manually.")
 
-    def read_definition_file(self, verify_branches: bool = True) -> None:
+    def read_definition_file(self, verify_branches: bool = True, cmd: str = None) -> None:
         with open(self._definition_file_path) as file:
             lines: List[str] = [line.rstrip() for line in file.readlines()]
 
         at_depth = {}
         last_depth = -1
-
+        high_lvl_cmds = {'traverse', 'status', 'discover'}
```

Comment:
Hmmm I'd rather NOT pollute `read_definition_file` with the logic to **decide** on whether to slide-out the invalid branches... I'd put this logic in cli.py instead

Reply:
@PawelLipski What should happen when we don't perform interactive slide-out, should we save the definition file or not?

Reply:
Personally, i would save the definition file, i see no reason to keep invalid branches.

Reply:
Sh*t, I didn't think of that... yeah, probably removing the invalid branch + printing a warning to **stderr** (not to pollute stdout) would be best